fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java

```diff
@@ -255,7 +256,9 @@ protected Pair<List<Long>, Long> calcActualSampleTablets(boolean forPartitionCol
         List<Long> sampleTabletIds = new ArrayList<>();
         long actualSampledRowCount = 0;
         boolean enough = false;
-        for (Partition p : olapTable.getPartitions()) {
+        List<Partition> sortedPartitions = olapTable.getPartitions().stream().sorted(
+                Comparator.comparing(Partition::getName)).collect(Collectors.toList());
+        for (Partition p : sortedPartitions) {
```

Comment:
do we need update inject stats of nereids tpcds shape test? @Jibing-Li

Reply:
Yes, we need to regenerate tpcds stats.

Reply:
Nereids tpcds shape test stats was collected using full analyze, so we don't need to update it.